Sources/RequestKit/JSONPostRouter.swift

```diff
@@ -5,7 +5,13 @@ import FoundationNetworking
 
 public protocol JSONPostRouter: Router {
     func postJSON<T>(_ session: RequestKitURLSession, expectedResultType: T.Type, completion: @escaping (_ json: T?, _ error: Error?) -> Void) -> URLSessionDataTaskProtocol?
-    func post<T: Codable>(_ session: RequestKitURLSession, decoder:JSONDecoder, expectedResultType: T.Type, completion: @escaping (_ json: T?, _ error: Error?) -> Void) -> URLSessionDataTaskProtocol?
+    func post<T: Codable>(_ session: RequestKitURLSession, decoder: JSONDecoder, expectedResultType: T.Type, completion: @escaping (_ json: T?, _ error: Error?) -> Void) -> URLSessionDataTaskProtocol?
+    
+    @available(macOS 12.0, iOS 15.0, tvOS 15.0, watchOS 8.0, *)
+    func postJSON<T>(_ session: RequestKitURLSession, expectedResultType: T.Type) async throws -> T?
+    
+    @available(macOS 12.0, iOS 15.0, tvOS 15.0, watchOS 8.0, *)
+    func post<T: Codable>(_ session: RequestKitURLSession, decoder: JSONDecoder, expectedResultType: T.Type) async throws -> T?
```

Comment:
I think these `async` functions should be enclosed with `#if swift(>=5.5) ~ #endif` to be ignored by non-latest Xcodes 🤔

Reply:
Doesn’t seem to work. Also `compiler(>=5.5)` seems to still trigger under Xcode 12.4